Source/bmalloc/bmalloc/darwin/BAllowProhibitedAPIs.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+
+#pragma once
+
+#include "BPlatform.h"
+
+#if !__has_include(<AvailabilityProhibitedInternal.h>)
+
+#if BPLATFORM(WATCHOS)
+#ifdef __WATCHOS_PROHIBITED
+#undef __WATCHOS_PROHIBITED
+#define __WATCHOS_PROHIBITED
+#endif
+#elif BPLATFORM(APPLETV)
+#ifdef __TVOS_PROHIBITED
+#undef __TVOS_PROHIBITED
+#define __TVOS_PROHIBITED
+#endif
+#elif BPLATFORM(IOS_FAMILY)
+#ifdef __IOS_PROHIBITED
+#undef __IOS_PROHIBITED
+#define __IOS_PROHIBITED
+#endif
```

Comment:
Not sure it matters, but this is slightly different than the logic above, which would redefine `__IOS_PROHIBITED` on macOS, too.